x-pack/solutions/search/plugins/search_homepage/public/hooks/use_is_sample_data_available.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { ILicense } from '@kbn/licensing-types';
+import useObservable from 'react-use/lib/useObservable';
+import { generateRandomIndexName } from '../utils/indices';
+import { useUserPrivilegesQuery } from './api/use_user_permissions';
+import { useKibana } from './use_kibana';
+
+export const useIsSampleDataAvailable = () => {
+  const { licensing, sampleDataIngest } = useKibana().services;
+
+  const indexName = useMemo(() => generateRandomIndexName(), []);
+  const { data: userPrivileges } = useUserPrivilegesQuery(indexName);
+  const license = useObservable<ILicense | null>(licensing.license$, null);
+  const hasLicense = useMemo(
+    () =>
+      sampleDataIngest?.minimumLicenseType
+        ? license?.hasAtLeast(sampleDataIngest?.minimumLicenseType)
```

Comment:
in `enterprise_search` we have these [utils](https://github.com/TattdCodeMonkey/kibana/blob/c2200263b5855d81e837ab405509a481d4ebba48/x-pack/solutions/search/plugins/enterprise_search/common/utils/licensing.ts) for checking the license in a uniform manner. Not sure how important the `isAvailable` & `isActive` checks are before calling `hasAtLeast`, I see lots of other usage without checking those so this is probably fine, but wanted to make you aware of those.

Reply:
Yep, makes sense to add it
Thank you